compiler/symbol.go

```diff
@@ -453,19 +456,32 @@ func isValidWasmType(typ types.Type, site wasmSite) bool {
 			return site == siteParam || site == siteIndirect
 		}
 	case *types.Array:
-		return site == siteIndirect && isValidWasmType(typ.Elem(), siteIndirect)
+		return site == siteIndirect && c.isValidWasmType(typ.Elem(), siteIndirect)
 	case *types.Struct:
 		if site != siteIndirect {
 			return false
 		}
+		// Structs with no fields do not need structs.HostLayout
+		if typ.NumFields() == 0 {
+			return true
+		}
+		var hasHostLayout bool
+		if !goenv.WantGoVersion(c.pkg.GoVersion(), 1, 23) {
+			hasHostLayout = true // package structs does not exist before go1.23
```

Comment:
I guess one way to work around this is to put this in a separate file with build tags. Together with https://github.com/tinygo-org/tinygo/pull/4501, the compiler won't try to compile projects with a Go version that's newer than the one TinyGo was compiled with (which means that when TinyGo was compiled with Go before 1.23, it will refuse to even compile code that requires Go 1.23+).

It's a bit ugly, but will keep compatibility for a bit longer.

Reply:
I can do this.

Reply:
@aykevl thanks! I rebased and my branch now uses `Parse`. I included the tests for `Parse` in my PR as well.